routing/payment_session.go

```diff
@@ -278,6 +278,12 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 				return nil, errNoPathFound
 			}
 
+			if !p.payment.DestFeatures.HasFeature(lnwire.MPPOptional) {
```

Comment:
Why isn't the above conditional being hit in practice? If we're paying a destination, that doesn't have a payment address in the invoice, then `PaymentAddr` will be set to `nil` meaning it should just fail as there's no payment address to use to correlate the payment splits.

Reply:
Or is this related to your wallet's custom implementation that uses the HTLC interceptor? As to my knowledge, in terms of base implementations, if they support `payment_addr`, then they're also able to receive MPP payments.

Reply:
Yeah, our impl supports `payment_addr` but not MPP. `payment_addr` was easy enough to implement and added a tangible security benefit, whereas mpp came at a high development cost so we left that for a future iteration. Is supporting both at the same time somehow expected? From the BOLTs I gathered they were developed closely but not that they were considered equal.

Reply:
Indeed, they were also split this way specifically because MPP has higher development costs